.github/workflows/_get_app_metadata.yml

```diff
@@ -0,0 +1,134 @@
+        run: pip install ledgered
+
+      - name: Gather application metadata, from inputs or 'ledger_app.toml'
+        id: fetch_metadata
+        run: |
+          pytest_directory='${{ inputs.pytest_directory }}';
+          if [[ -f "$APP_MANIFEST" ]];
+          then
+              # 'ledger_app.toml' exists
+              if grep -q 'rust-app' "$APP_MANIFEST";
+              then
+                  # legacy manifest -> Rust application
+                  echo "Legacy manifest detected. Please update your manifest to the newest version";
+                  # checking the manifest with the repo
+                  ledger-manifest --legacy --check . "$APP_MANIFEST";
+                  echo "build_directory=$(ledger-manifest --legacy --output-build-directory "$APP_MANIFEST")" >> "$GITHUB_OUTPUT";
+                  compatible_devices='${{ inputs.compatible_devices }}';
+                  echo "is_rust=true" >> "$GITHUB_OUTPUT";
+              else
+                  # classic manifest -> can be either C or Rust
+                  echo "Classic manifest detected.";
+                  # checking the manifest with the repo
+                  ledger-manifest --check . "$APP_MANIFEST";
+                  echo "build_directory=$(ledger-manifest --output-build-directory "$APP_MANIFEST")" >> "$GITHUB_OUTPUT";
+                  compatible_devices="$(ledger-manifest --output-devices "$APP_MANIFEST")";
+
+                  set +e  # when [tests.pytest_directory] is not set, ledger-manifest fails
+                  if temp="$(ledger-manifest --output-pytest-directory "$APP_MANIFEST")";
+                  then
+                      pytest_directory="${temp}";
+                  fi
+                  set -e
+
+                  if [[ "$(ledger-manifest --output-sdk "$APP_MANIFEST")" == "rust" ]];
+                  then
+                      echo "is_rust=true" >> "$GITHUB_OUTPUT";
+                  else
+                      echo "is_rust=false" >> "$GITHUB_OUTPUT";
+                  fi
+              fi
+          else
+              # No 'ledger_app.toml' -> C application
+              echo "No manifest detected.";
+              echo "build_directory=${{ inputs.relative_app_directory }}" >> "$GITHUB_OUTPUT";
+              compatible_devices='${{ inputs.compatible_devices }}';
+              echo "is_rust=false" >> "$GITHUB_OUTPUT";
+          fi
+
+          if [[ "${compatible_devices}" == 'None' ]];
+          then
+              # no inputs and no classic manifest
+              compatible_devices='["nanos", "nanosp", "nanox", "stax"]';
+          else
+              if [[ '${{ inputs.compatible_devices }}' != 'None' ]];
+              then
+                  # in case classic manifest with devices, the input takes precedence on it
+                  compatible_devices='${{ inputs.compatible_devices }}';
+              fi
+          fi
+
+          echo "compatible_devices=${compatible_devices}" | sed 's/+/p/' >> "$GITHUB_OUTPUT";
```

Comment:
We'll always struggle with the nanosp / nanos2 / nanos+  ...
If we use the nanosp here we avoid one conversion though

Reply:
My take is that we should push for `nanos+` wherever possible, then our tools / scripts make the conversion internally.
That's the official name of the product and we should hide our legacy naming / laziness to update from the front users.

Reply:
Ok